### PR TITLE
Add Github Actions CI, including release CI

### LINKED ISSUE #223: Improve our release process
Currently we don't have any automated release process: hpy releases (such as the upcoming 0.0.2) are manually uploaded using `twine`, and we upload only the `sdist`.

It would be nice to:
  - have an automated system which does a release whenever we push a relevant tag
  - build&upload wheels

### PR DESCRIPTION
**Pull request by dralley**
closes #223

### CODE REVIEW

**Comment by dralley** [comment]
TODO: artifact caching

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        python-version: ['3.9']   # ['2.7', '3.6', '3.7', '3.8', '3.9', '3.10.0-rc.1', 'pypy-2.7', 'pypy-3.6', 'pypy-3.7-nightly']
+        include:
+          - os: ubuntu-latest
+            python-version: '3.6'
+          - os: ubuntu-latest
+            python-version: '3.7'
+          - os: ubuntu-latest
+            python-version: '3.8'
+          - os: ubuntu-latest
+            python-version: '3.10.0-rc.1'
```
Comment (dralley) [inline .github/workflows/ci.yml:21]: See: https://github.com/dralley/hpy/pull/1/checks?check_run_id=3286253606

There appears to be a failure on 3.10 due to __slots__ changes

**Comment by dralley** [comment]
I'm going to close this and re-open it so I stop spamming your CI.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        include:
+          - os: ubuntu-latest
+            python-version: '3.6'
+          - os: ubuntu-latest
+            python-version: '3.7'
+          - os: ubuntu-latest
+            python-version: '3.8'
+          # Failure is expected on 3.10.0 currently
+          # - os: ubuntu-latest
+          #   python-version: '3.10.0-rc.1'
```
Comment (dralley) [inline .github/workflows/ci.yml:22]: See: https://github.com/dralley/hpy/pull/1/checks?check_run_id=3286253606

The legacy `__slots__` tests need to be disabled if Python version > 3.9

**Comment by dralley** [comment]
Tests are passing without caching https://github.com/dralley/hpy/actions/runs/1116661222

**Comment by antocuni** [comment]
@dralley thank you for your effort! I'm currently on vacation but I'll review this PR ASAP, unless some other core dev wants to do it first

**Comment by dralley** [comment]
@antocuni There's no rush, enjoy your vacation!

**Pull request closed without merging**